### ISSUE TITLE
Limit in MsSQL server is partially supported

### ISSUE THREAD

**Issue by skomlaebri** [ontop/ontop#225]
We currently use OFFSET and FETCH NEXT to translate limit, but this translation needs ORDER BY to work. See the problem with limit and mssql that has been reported here : #131.
The suggested workaround for the moment is to add, if possible, an order by.
Example:
```
SELECT ?item ?p
WHERE {
?item ?p http://my.test.com/mytesturi .
}
ORDER BY ?item
OFFSET 0 LIMIT 10
```

**Comment by chrdebru** [comment]
This problem is still relevant. I do not quite like the workaround. The error returned is "it.unibz.inf.ontop.exception.OntopQueryEvaluationException: com.microsoft.sqlserver.jdbc.SQLServerException: Incorrect syntax near 'OFFSET'." I would prefer that Ontop return an exception that a LIMIT without an ORDER for MS SQL servers is not supported.

**Comment by bcogrel** [comment]
Hi @chrdebru , thanks for reviving this issue. 

We also experienced this issue internally and we are working on a fix. Actually this kind of query can be supported by adding a "useless" ORDER BY, as suggested in https://stackoverflow.com/questions/30981059/how-to-use-offset-and-fetch-without-order-by-in-sql-server .

Best,
Benjamin

**Comment by chrdebru** [comment]
Hello Benjamin,
Thanks, but that solution only works partially in my case. Not only does it require a "ORDER BY (SELECT NULL) OFFSET 0" instead of "ORDER BY (SELECT NULL)" because I am working with views rather than tables. Because of that, the SQL generated for JOINS are subsequently invalid.

**Comment by chrdebru** [comment]
Hi. This solutions seems to work for tables. I have noticed that for views to play "nicely" with parentTriplesMaps, the queries must have a , "NULL AS FOO FROM tablename ORDER BY FOO OFFSET 0 ROWS" The OFFSET 0 Rows is important for views. When firing SPARQL queries, however the LIMIT needs an ORDER BY. The SQL query generated without the ORDER BY has an incorrect syntax.

**Comment by bcogrel** [comment]
Initial problem fixed  by https://github.com/ontop/ontop/commit/202ff7441a54faf45ba57c23d1888c5e1488bc60 .

@chrdebru , we haven't done anything particular for views. Could you please give a try to the latest version (available on Docker as `ontop/ontop-endpoint:4.1`)?

**Comment by ghxiao** [comment]
Feel free to reopen if the problem persists
